gammapy/datasets/core.py

```diff
@@ -222,7 +222,7 @@ def read(cls, path, filedata="_datasets.yaml", filemodel="_models.yaml"):
             datasets.append(dataset)
         return cls(datasets)
 
-    def write(self, path, prefix="", overwrite=False):
+    def write(self, path, prefix="", overwrite=False, save_covariance=True):
```

Comment:
Nitpicky: maybe rename the option to `write_covariance` so that it is in line with the name of the corresponding method on the `Models` class.